### PR TITLE
Added new handwired keyboard Wakizashi 40

### PR DESCRIPTION
**Pull request by xia0**
## Wakizashi 40 keyboard

Added keyboard for a 3d printed hand wired keyboard I designed. It's available freely on thingiverse - see readme.
This is my first contribution so please let me know if anything is not right.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by drashna — APPROVED** [review]

**Review by zvecr — COMMENTED** [review]
As you are using data driven info.json, you can actually remove `keyboards/handwired/wakizashi40/wakizashi40.c` and `keyboards/handwired/wakizashi40/wakizashi40.h`.

**Comment by xia0** [comment]
> As you are using data driven info.json, you can actually remove `keyboards/handwired/wakizashi40/wakizashi40.c` and `keyboards/handwired/wakizashi40/wakizashi40.h`.

Thank you. Removed these files. In config.h I've got `#define DYNAMIC_KEYMAP_LAYER_COUNT 12` - is there a way to define this in info.json?

File: keyboards/handwired/wakizashi40/readme.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+Reverse stagger 40% keyboard based on the Katana 60.
+
+* Keyboard Maintainer: [fumbucker](https://github.com/xia0)
+* Hardware Supported: Teensy2++
+* Hardware Availability: [thingiverse](https://www.thingiverse.com/thing:3467930)
+
+Make example for this keyboard (after setting up your build environment):
+
+    qmk compile -kb handwired/wakizashi40 -km via
```
Comment (fauxpark) [inline keyboards/handwired/wakizashi40/readme.md:13]:
```suggestion
    make handwired/wakizashi40:default
```

File: keyboards/handwired/wakizashi40/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+/*
+[8:10 PM] я лов днкс: hey i was wondering if anyone could help me understand how to get a board onto VIA
+[8:11 PM] я лов днкс: from what I gather, i have to get it into the qmk repo then wait for the via team to update?
+[8:13 PM] sigprof: yes, first you need to make a PR to https://github.com/qmk/qmk_firmware with proper keyboard support, including the via keymap; then you make a PR with the VIA JSON to https://github.com/the-via/keyboards, providing a link to the QMK PR (there is a place for it in the template).
+[8:15 PM] я лов днкс: ty ty
+[8:17 PM] sigprof: also https://get.vial.today/docs/porting-to-via.html may be useful, even if you don't plan to use the Vial extensions
```
Comment (fauxpark) [inline keyboards/handwired/wakizashi40/keymaps/default/keymap.c:21]:
```suggestion
```

File: keyboards/handwired/wakizashi40/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+		KC_LSFT, KC_Z, KC_X, KC_C, KC_V, KC_B, KC_ESC, KC_N, KC_M, KC_COMM, KC_DOT, KC_SLSH, KC_RSFT,
+		KC_LCTL, KC_LGUI, KC_LALT, LT(1, KC_SPC), LT(2, KC_SPC), KC_LALT, KC_MENU, KC_LCTL),
+
+	[1] = LAYOUT_all(
+		KC_F1, KC_F2, KC_F3, KC_F4, KC_F5, KC_F5, KC_F7, KC_F8, KC_F9, KC_F10, KC_F11, KC_F12,
+		MO(11), KC_1, KC_2, KC_3, KC_4, KC_5, KC_6, KC_7, KC_8, KC_9, KC_0, KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_MINS, KC_TRNS, KC_EQL, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_SPC, KC_SPC, KC_TRNS, KC_TRNS, KC_TRNS),
+
+	[2] = LAYOUT_all(
```
Comment (fauxpark) [inline keyboards/handwired/wakizashi40/keymaps/default/keymap.c:39]: Please remove all these useless layers.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/handwired/wakizashi40/keymaps/via/keymap.c [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+/*
+[8:10 PM] я лов днкс: hey i was wondering if anyone could help me understand how to get a board onto VIA
+[8:11 PM] я лов днкс: from what I gather, i have to get it into the qmk repo then wait for the via team to update?
+[8:13 PM] sigprof: yes, first you need to make a PR to https://github.com/qmk/qmk_firmware with proper keyboard support, including the via keymap; then you make a PR with the VIA JSON to https://github.com/the-via/keyboards, providing a link to the QMK PR (there is a place for it in the template).
+[8:15 PM] я лов днкс: ty ty
+[8:17 PM] sigprof: also https://get.vial.today/docs/porting-to-via.html may be useful, even if you don't plan to use the Vial extensions
```
Comment (fauxpark) [inline keyboards/handwired/wakizashi40/keymaps/via/keymap.c:21]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/handwired/wakizashi40/keymaps/default/keymap.c (hunk 3 above)
Comment (xia0) [reply keyboards/handwired/wakizashi40/keymaps/default/keymap.c:32]: Will this cause issues with having `#define DYNAMIC_KEYMAP_LAYER_COUNT 12` set in config.h? If I leave the extra layers in the via keymap it should work without issue?

**Review by fauxpark — APPROVED** [review]

File: keyboards/handwired/wakizashi40/keymaps/default/keymap.c (hunk 3 above)
Comment (fauxpark) [reply keyboards/handwired/wakizashi40/keymaps/default/keymap.c:32]: VIA should not be enabled on the default keymap, so the layer count matching the define is irrelevant here.

**Pull request merged**